src/pyggdrasil/tree_inference/_mcmc.py

```diff
@@ -402,6 +411,9 @@ def _mcmc_kernel(
     else:
         proposal, log_q_diff = _swap_subtrees_proposal(key, tree)
 
+    # reorder tree, for 0-N indexing for log-probability function
+    proposal = OrderedTree.from_tree(proposal)
+
```

Comment:
The FIX! - Regardless of the move I reorder the proposed tree in the MCMC kernel now.

Reply:
Yes - they should as tree generation always gives ordered trees. 

Now that I consider this again, it might be violated by HUNTRESS. 

I added a reordering in the MCMC sampler. 

Well spotted!